### PR TITLE
Add support for Tekton Tasks

### PR DESCRIPTION
**Pull request by geetikakay**
##### Short description:
Add support for tekton tasks : https://tekton.dev/docs/pipelines/tasks/#configuring-a-task

##### What this PR does / why we need it:
This PR is needed to support cluster_resouce task .

## Summary by CodeRabbit

- **New Features**
	- Introduced a `Task` class to enhance task configuration and management within continuous integration workflows in Kubernetes.
	- Added functionality for users to define complex task specifications, including steps, parameters, and resources.

- **Bug Fixes**
	- Implemented validation to ensure required task attributes are provided, improving error handling.

### CODE REVIEW

**Comment by msi-b1** [comment]
Report bugs in [Issues](https://github.com/myakove/github-webhook-server/issues)

The following are automatically added:
 * Add reviewers from OWNER file (in the root of the repository) under reviewers section.
 * Set PR size label.
 * New issue is created for the PR. (Closed when PR is merged/closed)
 * Run [pre-commit](https://pre-commit.ci/) if `.pre-commit-config.yaml` exists in the repo.

Available user actions:
 * To mark PR as WIP comment `/wip` to the PR, To remove it from the PR comment `/wip cancel` to the PR.
 * To block merging of PR comment `/hold`, To un-block merging of PR comment `/hold cancel`.
 * To mark PR as verified comment `/verified` to the PR, to un-verify comment `/verified cancel` to the PR.
        verified label removed on each new commit push.
 * To cherry pick a merged PR comment `/cherry-pick <target branch to cherry-pick to>` in the PR.
    * Multiple target branches can be cherry-picked, separated by spaces. (`/cherry-pick branch1 branch2`)
    * Cherry-pick will be started when PR is merged
 * To build and push container image command `/build-and-push-container` in the PR (tag will be the PR number).
    * You can add extra args to the Podman build command
        * Example: `/build-and-push-container --build-arg OPENSHIFT_PYTHON_WRAPPER_COMMIT=<commit_hash>`
 * To add a label by comment use `/<label name>`, to remove, use `/<label name> cancel`
 * To assign reviewers based on OWNERS file use `/assign-reviewers`
 * To check if PR can be merged use `/check-can-merge`

<details>
<summary>Supported /retest check runs</summary>

 * `/retest tox`: Retest tox
 * `/retest python-module-install`: Retest python-module-install

</details>

<details>
<summary>Supported labels</summary>

 * hold
 * verified
 * wip
 * lgtm

</details>

**Review by rnetser — CHANGES_REQUESTED** [review]
Please use https://github.com/RedHatQE/openshift-python-wrapper/blob/main/class_generator/README.md to generate the module

File: ocp_resources/task.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
+from ocp_resources.resource import MissingRequiredArgumentError, NamespacedResource
```
Comment (myakove) [inline ocp_resources/task.py:1]: Did you use the script to generate the class?

Comment (geetikakay) [reply ocp_resources/task.py:1]: No I will use it. Thanks for pointing

Comment (geetikakay) [reply ocp_resources/task.py:1]: @myakove  maybe i missed something but i see this issue https://github.com/RedHatQE/openshift-python-wrapper/issues/2006

File: ocp_resources/task.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        self.results = results
+        self.volumes = volumes
+        self.step_template = step_template
+        self.sidecars = sidecars
+
+    def to_dict(self) -> None:
+        super().to_dict()
+        if not self.yaml_file:
+            self.res["spec"] = {}
+            self.res["spec"]["steps"] = self.steps
```
Comment (rnetser) [inline ocp_resources/task.py:56]:
```suggestion
            _spec = self.res["spec"]
            _spec= {"steps": self.steps}
```

and re-use `_spec`

File: ocp_resources/task.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+            description (str, optional): An informative description of the Task.
+            params (list, optional): Specifies execution parameters for the Task.
+            workspaces (list, optional): Specifies paths to volumes required by the Task.
+            results (list, optional): Specifies the names under which Tasks write execution results.
+            volumes (list, optional): Specifies one or more volumes that will be available to the Steps in the Task.
+            step_template (dict, optional): Specifies a Container step definition to use as the basis for all Steps in the Task.
+            sidecars (list, optional): Specifies Sidecar containers to run alongside the Steps in the Task.
+        """
+        if not steps:
+            raise MissingRequiredArgumentError(argument="'steps' is required")
```
Comment (rnetser) [inline ocp_resources/task.py:39]: only arg name should be passed

File: ocp_resources/task.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        volumes: Optional[List[Dict[str, Dict[str, Any]]]] = None,
+        step_template: Optional[Dict[str, Any]] = None,
+        sidecars: Optional[List[Dict[str, Any]]] = None,
+        **kwargs,
+    ):
+        """
+        Create and manage Task which specifies a sequence of steps to be executed.
+
+        Args:
+            steps (list): Specifies one or more container images to run in the Task.
```
Comment (rnetser) [inline ocp_resources/task.py:29]: please update with correct typing (e.g List[Dict...])

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ocp_resources/task.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+            steps (List[Dict[str, Any]]): Specifies one or more container images to run in the Task.
+            description (Optional[str]): An informative description of the Task.
+            params (Optional[List[Dict[str, str]]]): Specifies execution parameters for the Task.
+            workspaces (Optional[List[Dict[str, Any]]]): Specifies paths to volumes required by the Task.
+            results (Optional[List[Dict[str, Any]]]): Specifies the names under which Tasks write execution results.
+            volumes (Optional[List[Dict[str, Dict[str, Any]]]]): Specifies one or more volumes that will be available to the Steps in the Task.
+            step_template (Optional[Dict[str, Any]]): Specifies a Container step definition to use as the basis for all Steps in the Task.
+            sidecars (Optional[List[Dict[str, Any]]]): Specifies Sidecar containers to run alongside the Steps in the Task.
+        """
+        if not steps:
```
Comment (myakove) [inline ocp_resources/task.py:38]: Should be under `to_dict`, user should be able to use this resource for `get` without pass `steps`

**Review by myakove — CHANGES_REQUESTED** [review]

**Comment by myakove** [comment]
@coderabbitai review

**Review by myakove — APPROVED** [review]

**Review by rnetser — APPROVED** [review]

**Review by myakove — APPROVED** [review]

**Pull request merged**